### PR TITLE
Add support for is_null and is_not_null operators

### PR DESCRIPTION
**Pull request by jimf**

### CODE REVIEW

File: query-builder-elasticsearch.js [hunk 1]
```diff
@@ -108,6 +110,8 @@
                         // we express this with a sub boolean query and must_not.
                         if (data.condition === 'OR' && (rule.operator === 'not_equal' || rule.operator === 'not_in')) {
                             return {'bool': {'must_not': [part]}}
+                        } else if (rule.operator === 'is_null' || rule.operator === 'is_not_null') {
```
Comment (piotch) [inline query-builder-elasticsearch.js]: I think that this causes a bug when you do use a OR condition with one of the branches that uses the is_null (must not exist) operator. I made a test to show this.

```
QUnit.test("OR condition and null op", function (assert) {

        $b.queryBuilder({
            filters: basic_filters,
            rules: {
                condition: 'OR',
                rules: [
                    {id: 'name', field: 'name', operator: 'is_not_null'},
                    {id: 'price', field: 'price', operator: 'is_null'}
                ]
            }
        });

        assert.deepEqual(
            $b.queryBuilder('getESBool'),
            {"bool": {"should":[
                {"exists":{"field":"name"}},
                {"bool":{"must_not": {"exists": {"field":"price"}}}}
            ]}},
            'Should build a should query with a bool subquery'
        );

    });
```

**Comment by piotch** [comment]
Hello I added a comment, I think that there is a case (OR query with is_null in a branch) that is not generated correctly. I propose to add the test below, let me know if you think this is correct. 

```
    QUnit.test("OR condition and null op", function (assert) {

        $b.queryBuilder({
            filters: basic_filters,
            rules: {
                condition: 'OR',
                rules: [
                    {id: 'name', field: 'name', operator: 'is_not_null'},
                    {id: 'price', field: 'price', operator: 'is_null'}
                ]
            }
        });

        assert.deepEqual(
            $b.queryBuilder('getESBool'),
            {"bool": {"should":[
                {"exists":{"field":"name"}},
                {"bool":{"must_not": {"exists": {"field":"price"}}}}
            ]}},
            'Should build a should query with a bool subquery'
        );

    });
```

**Review by piotch — CHANGES_REQUESTED** [review]

File: query-builder-elasticsearch.js (hunk 1 above)
Comment (piotch) [reply query-builder-elasticsearch.js]: This said, do not hesitate to tell me if this incorrect, maybe there are different ways to translate an OR query.

Comment (jimf) [reply query-builder-elasticsearch.js]: Ah, yes, you're correct. Good catch. I'll update.

**Comment by jimf** [comment]
@piotch Updated

**Review by piotch — APPROVED** [review]

**Comment by piotch** [comment]
Great, thank you!

**Pull request merged**